tests/Support/SupportCollectionTest.php

```diff
@@ -895,6 +895,35 @@ public function testFlatMap()
         $this->assertEquals(['programming', 'basketball', 'music', 'powerlifting'], $data->all());
     }
 
+    public function testMapToAssoc()
+    {
+        $data = new Collection([
+            ['name' => 'Blastoise', 'type' => 'Water', 'idx' => 9],
+            ['name' => 'Charmander', 'type' => 'Fire', 'idx' => 4],
+            ['name' => 'Dragonair', 'type' => 'Dragon', 'idx' => 148],
+        ]);
+        $data = $data->mapToAssoc(function ($pokemon) {
+            return [$pokemon['name'] => $pokemon['type']];
+        });
+        $this->assertEquals(
+            ['Blastoise' => 'Water', 'Charmander' => 'Fire', 'Dragonair' => 'Dragon'],
+            $data
+        );
+    }
+
+    public function testToAssoc()
+    {
+        $collection = new Collection([
+            ['Blastoise' => 'Water'],
+            ['Charmander' => 'Fire'],
+            ['Dragonair' => 'Dragon'],
+        ]);
+        $this->assertEquals(
+            ['Blastoise' => 'Water', 'Charmander' => 'Fire', 'Dragonair' => 'Dragon'],
+            $collection->toAssoc()
+        );
+    }
```

Comment:
I think we should just have `mapToAssoc` call `flatMap` directly. Let it be an alias for now.

We'll add tests to `mapToAssoc` making sure it actually maps the non-numeric keys properly.

Then, later on, if the implementations ever need to diverge we can let them diverge.